discovery/gossiper.go

```diff
@@ -814,126 +809,69 @@ func (d *deDupedAnnouncements) Emit() []msgWithSenders {
 // will try to resend them. If we have the full proof, we can safely delete the
 // message from the messageStore.
 func (d *AuthenticatedGossiper) resendAnnounceSignatures() error {
-	type msgTuple struct {
-		peer  *btcec.PublicKey
-		msg   *lnwire.AnnounceSignatures
-		dbKey []byte
+	peerMsgsToResend, err := d.cfg.MessageStore.Messages()
+	if err != nil {
+		return err
 	}
 
-	// Fetch all the AnnounceSignatures messages that was added to the
-	// database.
-	//
-	// TODO(halseth): database access should be abstracted
-	// behind interface.
-	var msgsResend []msgTuple
-	if err := d.cfg.DB.View(func(tx *bbolt.Tx) error {
-		bucket := tx.Bucket(messageStoreKey)
-		if bucket == nil {
-			return nil
+	// We now iterate over these messages, resending those that we don't
+	// have the full proof for, deleting the rest.
+	for peer, msgsToResend := range peerMsgsToResend {
+		pubKey, err := btcec.ParsePubKey(peer[:], btcec.S256())
+		if err != nil {
+			return err
 		}
 
-		if err := bucket.ForEach(func(k, v []byte) error {
-			// The database value represents the encoded
-			// AnnounceSignatures message.
-			r := bytes.NewReader(v)
-			msg := &lnwire.AnnounceSignatures{}
-			if err := msg.Decode(r, 0); err != nil {
-				return err
-			}
+		for _, msg := range msgsToResend {
+			msg := msg.(*lnwire.AnnounceSignatures)
```

Comment:
No type assertion to ensure it's actually an `AnnounceSignatures`. Should also be a type switch since this can now have other message types.

Reply:
At this point, it's not possible for other messages to exist within it. In the follow-up PR (#1595) most of this diff will be removed/modified.